src/Workspaces/CoreTest/WorkspaceTests/WorkspaceTests.cs

```diff
@@ -55,18 +44,7 @@ public void TestChangeDocumentName_TryApplyChanges_Throws()
             var changedDoc = originalDoc.WithName(newName);
             Assert.Equal(newName, changedDoc.Name);
 
-            NotSupportedException e = null;
-            try
-            {
-                ws.TryApplyChanges(changedDoc.Project.Solution);
-            }
-            catch (NotSupportedException x)
-            {
-                e = x;
-            }
-
-            Assert.NotNull(e);
-            Assert.Equal(WorkspacesResources.Changing_document_property_is_not_supported, e.Message);
+            AssertEx.Throws<NotSupportedException>(WorkspacesResources.Changing_document_property_is_not_supported, () => ws.TryApplyChanges(changedDoc.Project.Solution));
```

Comment:
So the existing xUnit Assert.Throws returns the exception, so you didn't really need the new helper.